### PR TITLE
fixing the disclosure indicator position

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
This is PR for fix the disclosure  indicator position. Now it's aligned under the separators.
## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4239

## Screenshots
IMG_0018

IMG_0019

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 1]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
+		
+		let backgroundView: UIView
+		if let existingBackgroundView = self.backgroundView {
+			backgroundView = existingBackgroundView
+		} else {
+			backgroundView = UIView()
+			self.backgroundView = backgroundView
+		}
+		backgroundView.addSubview(separator)
+
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:305]:
```suggestion
```
Remove that strange background logic

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 2]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
+			backgroundView = UIView()
+			self.backgroundView = backgroundView
+		}
+		backgroundView.addSubview(separator)
+		
 		separator.backgroundColor = .enaColor(for: .hairline)
 		separator.translatesAutoresizingMaskIntoConstraints = false
-		separator.trailingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.trailingAnchor).isActive = true
+		separator.leadingAnchor.constraint(equalTo: backgroundView.leadingAnchor, constant: 16).isActive = true
+		separator.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor, constant: -16).isActive = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:309]:
```suggestion
		separator.leadingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leadingAnchor, constant: -2.0).isActive = true
		separator.trailingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.trailingAnchor, constant: 32.0).isActive = true
```
Use still contentView for layout

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 3]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
 		separator.heightAnchor.constraint(equalToConstant: 1).isActive = true
 
 		switch location {
 		case .top:
 			separator.tag = SeparatorLocation.top.rawValue
-			separator.topAnchor.constraint(equalTo: contentView.topAnchor).isActive = true
-			separator.leadingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leadingAnchor).isActive = true
+			separator.topAnchor.constraint(equalTo: backgroundView.topAnchor).isActive = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:315]:
```suggestion
			separator.topAnchor.constraint(equalTo: contentView.topAnchor).isActive = true
```

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 4]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
 		case .bottom:
 			separator.tag = SeparatorLocation.bottom.rawValue
-			separator.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).isActive = true
-			separator.leadingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leadingAnchor).isActive = true
+			separator.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor).isActive = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:318]:
```suggestion
			separator.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).isActive = true
```

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 5]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
 		case .inBetween:
 			separator.tag = SeparatorLocation.inBetween.rawValue
-			separator.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).isActive = true
-			separator.leadingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leadingAnchor).isActive = true
+			separator.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor).isActive = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:321]:
```suggestion
			separator.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).isActive = true
```

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:305]: Thanks Kai. It's removed.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift (hunk 2 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:309]: revert it.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift (hunk 3 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:315]: Done.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift (hunk 4 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:318]: Done

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift (hunk 5 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:321]: Thanks, Done.

**Review by KaiTeuber — APPROVED** [review]
Looks much better now 👍

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift [hunk 6]
```diff
@@ -293,25 +293,32 @@ private extension UITableViewCell {
...
-		contentView.addSubview(separator)
+
+		let seperatorView: UIView
+		if let existingSeperatorView = self.backgroundView {
+			seperatorView = existingSeperatorView
+		} else {
+			seperatorView = UIView()
+			self.backgroundView = seperatorView
+		}
+		seperatorView.addSubview(separator)
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:304]: I still think this is very hard to read

Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewController.swift:304]: Yes, but no other solution works at the moment.

**Review by KaiTeuber — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**